oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/name/ReadOnlyNamespaceRegistry.java

```diff
@@ -112,4 +120,53 @@ public String getPrefix(String uri) throws NamespaceException {
                 "No namespace prefix registered for URI " + uri);
     }
 
+    protected void checkConsistency() {
+        final String jcrPrimaryType = "jcr:primaryType";
+        List<String> prefixes = Arrays.asList(getPrefixes());
+        List<String> uris = Arrays.asList(getURIs());
+        if (prefixes.size() != uris.size()) {
+            LOG.error("The namespace registry is inconsistent: found {} registered prefixes and {} registered URIs. The numbers have to be equal.", prefixes.size(), uris.size());
```

Comment:
In the log messages, I'd replace "URI" by either "namespace name" or "namespace URI", and "prefix" to "namespace prefix".